lib/middleware/tagged_logging.rb

```diff
@@ -20,5 +22,14 @@ def call(env)
       ::Logging.mdc.delete('request')
       ::Logging.mdc.delete('session')
     end
+
+    private
+
+    # Prefer using a digest of the session ID so it can be passed safely to external smart proxies
+    # etc to prevent session hijacks
+    def digest_id(id)
+      salt = defined?(EncryptionKey::ENCRYPTION_KEY) ? EncryptionKey::ENCRYPTION_KEY : 'default'
+      OpenSSL::HMAC.hexdigest(OpenSSL::Digest.new('sha256'), salt, id)
+    end
```

Comment:
Yes, fair point. I've replaced it with a generated ID, no need to base it on the session ID or hash it.